File: src/main/kotlin/school/lec02/Lec02Main.java
package school.lec02;

public class Lec02Main {

    public static void main(String[] args) {

    }

    public boolean startsWithA1(String str){
        if(str==null){
            throw new RuntimeException("null이 들어왔습니다.");
        }
        return str.startsWith("A");
    }

    public Boolean startsWithA2(String str){
        if(str==null){
            return null;
        }
        return str.startsWith("A");
    }

    public boolean startWithA3(String str){
        if(str==null) {
            return false;
        }
        return str.startsWith("A");
    }

}

File: src/main/kotlin/school/lec02/Lec02Main.kt
package school.lec02

import java.lang.RuntimeException

fun main(){

    val number1: Int? = null
    val number2: Long = number1?.toLong() ?: 0L

    print(number2)

}

fun startsWithA1(str: String?): Boolean {
    return str?.startsWith("A")
        ?: throw RuntimeException("NULL 입니다.")
}

fun startsWith2(str: String?): Boolean? {
    if(str==null){
        return null
    }
    return str.startsWith("A")
}

fun startsWith3(str: String?): Boolean {
    if(str==null){
        return false
    }
    return str.startsWith("A")
}

fun startsWithA5(str: String?): Boolean{
    return str?.startsWith("A")
        ?: throw RuntimeException("null 입니다.")
}

fun startsWithA6(str: String?): Boolean? {
    return str?.startsWith("A")
}

fun startsWithA7(str: String?): Boolean{
    return str?.startsWith("A") ?: false
}

fun startsWithA9(str: String?): Boolean{
    return str!!.startsWith("A")
}